src/Auth.test.ts

```diff
@@ -15,6 +15,28 @@ it("should register second user", async () => {
   ).toResolve();
 });
 
+it("should register a magic link user when magic link mode is enabled", async () => {
+  await nhost.withEnv({
+    ENABLE_MAGIC_LINK: 'true',
+  }, async () => {
+    await expect(
+      auth.register({ email: "magic-link-user@nhost.io" })
+    ).toResolve();
+  }, {
+    ENABLE_MAGIC_LINK: 'false'
+  })
+});
+
+it("should not register a magic link user when magic link mode is disabled", async () => {
+  await nhost.withEnv({
+    ENABLE_MAGIC_LINK: 'false',
+  }, async () => {
+    await expect(
+      auth.register({ email: "magic-link-user@nhost.io" })
+    ).toReject();
+  })
+});
```

Comment:
Do we also want a test where the `password` is empty? 

Let's say the user should fill in a form to register with email and password, and only fill in `email` and click "Register". How will this be handled by the js sdk? Will the user "accidently" register using a magic link? PReferably the js sdk should throw an error.

Should work:

```js
const email = 'johan@nhost.io';
nhost.auth.register({email});
```

Should fail:
```js
const email = 'johan@nhost.io';
const password = '';
nhost.auth.register({email, password});
```

```js
const email = 'johan@nhost.io';
const password;
nhost.auth.register({email, password});
```